Add tip about using local instead of locals for complex cases (#11036)
hashicorp/packer: website/content/docs/templates/hcl_templates/locals.mdx

}
```

## Single `local` block

The `local` block defines exactly one local variable within a folder. The block
label is the name of the local, and the "expression" is the expression that
should be evaluated to create the local. Using this block, you can optionally
supply a "sensitive" boolean to mark the variable as sensitive and filter it
from logs.

```hcl
local "mylocal" {
  expression = "${var.secret_api_key}"
  sensitive  = true
}
```

> This block is also very useful to define complex locals. Packer might take some time to expands and evaluate `locals`

Reviewer: ```suggestion
This block is also very useful for defining complex locals. Packer might take some time to expand and evaluate `locals`
```

> with complex expressions dependent on other locals. The `locals` block is read as map and maps are not sorted,

Reviewer: ```suggestion
with complex expressions dependent on other locals. The `locals` block is read as a map.  Maps are not sorted, and therefore
```

> and the evaluation time is not deterministic.

Reviewer 2: ```suggestion
the evaluation time is not deterministic.
```

> To avoid that, singular `local` block should be used instead. These will be

Reviewer: ```suggestion
To avoid that, singular `local` blocks should be used instead. These will be
```
